src/librustc/hir/lowering.rs

```diff
@@ -1865,20 +1914,32 @@ impl<'a> LoweringContext<'a> {
         )
     }
 
-    fn lower_local(&mut self, l: &Local) -> P<hir::Local> {
+    fn lower_local(&mut self, l: &Local) -> (P<hir::Local>, OneVector<hir::ItemId>) {
         let LoweredNodeId { node_id, hir_id } = self.lower_node_id(l.id);
-        P(hir::Local {
+        let mut ids = OneVector::<hir::ItemId>::new();
+        if let Some(ref ty) = l.ty {
+            let mut visitor = ImplTraitTypeIdVisitor { ids: &mut ids };
+            visitor.visit_ty(ty);
+        }
+        let impl_trait_owner_id = *self.current_impl_trait_owner.last().unwrap();
+        (P(hir::Local {
             id: node_id,
             hir_id,
             ty: l.ty
                 .as_ref()
-                .map(|t| self.lower_ty(t, ImplTraitContext::Disallowed)),
+                .map(|t| self.lower_ty(t,
+                    if self.sess.features_untracked().impl_trait_in_bindings {
+                        ImplTraitContext::Existential(Some(impl_trait_owner_id))
```

Comment:
@alexreg  `impl_trait_owner_id` is wrapped into `Option` here.

Reply:
Ah, my bad. Fair point then!

Reply:
On revisiting this, @csmoe may in fact have a good point. I think we need to allow for the `Option<DefId>` being `None` here for the case of locals defined within block expressions for statics & consts. Maybe you can clarify this however, @cramertj @oli-obk?